equistore-torch/src/labels.cpp

```diff
@@ -158,60 +159,272 @@ TorchLabels LabelsHolder::create(
     );
 }
 
+
+LabelsHolder::LabelsHolder(std::vector<std::string> names, torch::Tensor values, CreateView):
+    names_(std::move(names)),
+    values_(std::move(values))
+{}
+
+TorchLabels LabelsHolder::view(const TorchLabels& labels, std::vector<std::string> names) {
+    if (names.empty()) {
+        C10_THROW_ERROR(ValueError,
+            "can not index Labels with an empty list of dimension names"
+        );
+    }
+
+    auto dimensions = std::vector<int64_t>();
+    for (const auto& name: names) {
+        auto it = std::find(std::begin(labels->names_), std::end(labels->names_), name);
+        if (it == std::end(labels->names_)) {
+            C10_THROW_ERROR(ValueError,
+                "'" + name + "' not found in the dimensions of these Labels"
+            );
+        }
+
+        auto index = std::distance(std::begin(labels->names_), it);
+        dimensions.push_back(static_cast<int64_t>(index));
+    }
+
+    auto new_values = labels->values_.index({torch::indexing::Slice(), torch::tensor(std::move(dimensions))});
+    return torch::make_intrusive<LabelsHolder>(std::move(names), std::move(new_values), CreateView{});
+}
+
 LabelsHolder::LabelsHolder(equistore::Labels labels):
     names_(names_from_equistore(labels)),
     values_(values_from_equistore(labels)),
     labels_(std::move(labels))
 {}
 
+const equistore::Labels& LabelsHolder::as_equistore() const {
+    if (!labels_.has_value()) {
+        C10_THROW_ERROR(ValueError,
+            "can not call this function on Labels view, call to_owned first"
+        );
+    }
+
+    return labels_.value();
+}
+
+LabelsHolder LabelsHolder::to_owned() const {
```

Comment:
Not exactly sure what this does. So `has_value` checks if is a "None" type (part from Optional type)? If it is not "None", then return itself. Otherwise it creates an LabelHolder with the names and values. I am not sure when this else case is used.

Reply:
If `LabelsHolder::labels_` is an `equistore::Labels`, we return that. If `LabelsHolder::labels_` is a nullopt (i.e. None), then we create a new equistore::Labels, and a new LabelsHolder wrapping it. 

`LabelsHolder::labels_` will be a nullopt when we create a column view for some Labels, with something like `labels[["center", "structure"]]`

I'll add some more explanations around this